src/lib/operators/sort.cpp

```diff
@@ -1,29 +1,26 @@
 #include "sort.hpp"
 
-#include <functional>
-#include <memory>
-#include <string>
-#include <unordered_map>
-#include <utility>
-#include <vector>
-
-#include "storage/reference_segment.hpp"
-#include "storage/segment_accessor.hpp"
 #include "storage/segment_iterate.hpp"
-#include "storage/value_segment.hpp"
 
 namespace opossum {
 
-Sort::Sort(const std::shared_ptr<const AbstractOperator>& in, const ColumnID column_id, const OrderByMode order_by_mode,
-           const size_t output_chunk_size)
+Sort::Sort(const std::shared_ptr<const AbstractOperator>& in,
+                 const std::vector<SortColumnDefinition>& sort_definitions, const size_t output_chunk_size)
     : AbstractReadOnlyOperator(OperatorType::Sort, in),
-      _column_id(column_id),
-      _order_by_mode(order_by_mode),
-      _output_chunk_size(output_chunk_size) {}
+      _sort_definitions(sort_definitions),
+      _output_chunk_size(output_chunk_size) {
+  _sort_definition_data_types.resize(_sort_definitions.size());
+  for (auto column_id = ColumnID{0}; column_id < _sort_definitions.size(); ++column_id) {
+    auto sort_definition = _sort_definitions[column_id];
+    _sort_definition_data_types[column_id] = input_table_left()->column_data_type(sort_definition.column);
```

Comment:
I'm not sure if we should access `input_table_left()` here. Two tests are failing independently because of a bad access exception – apparently ``input_table_left()` returns `nullptr`.

Reply:
Resolved in 15dc255

Reply:
Resolved in 15dc255